src/core/database.py

```diff
@@ -20,7 +42,42 @@ def __len__(self):
         return len(self.trajectory)
 
     def get_rgb_image_by_index(self, n: int) -> NDArray[Shape["*, *, 3"], UInt8]:
-        return self.images[n].read()
+        return self.images[n].image
 
     def get_pcd_by_index(self, n: int) -> o3d.geometry.PointCloud:
-        return self.pcds[n].read()
+        return self.pcds[n].point_cloud
+
+    def build_sparse_map_with_caching(
+        self,
+        voxel_grid: VoxelGrid,
+        down_sample_step: int = 100,
+    ) -> o3d.t.geometry.PointCloud:
+        """
+        Builds sparse map of the whole DB scene
+        :param voxel_grid: Voxel grid for down sampling
+        :param down_sample_step: Voxel down sampling step for reducing RAM usage
+        :return: Resulting point cloud of the scene
+        """
+        return _build_sparse_map_with_caching(
+            self.trajectory, self.pcds, voxel_grid, down_sample_step
+        )
+
+    @cached_property
+    def bounds(
```

Comment:
do we need to calculate it separate from map? maybe we can save this info during map calc too?